app/models/generic_object_definition.rb

```diff
@@ -7,6 +16,21 @@ class GenericObjectDefinition < ApplicationRecord
   has_many  :generic_objects
 
   def defined_attributes
-    @defined_attributes ||= properties[:attributes].stringify_keys
+    properties[:attributes]
+  end
+
+  def properties=(props)
+    props = props.symbolize_keys
+    if props.key?(:attributes)
+      props[:attributes] = props[:attributes].each_with_object({}) do |(name, type), hash|
+        raise ArgumentError, "#{type} is not a recognized type" unless TYPE_MAP.key?(type.to_sym)
+        hash[name.to_s] = type.to_sym
+      end
+    end
```

Comment:
Good point...I forgot props already was duped...

yeah, I'm fine with your implementation, I was just seeing if an alternate one might be better/easier to read.